projectm: 3.1.12 -> 4.1.2 (split into libprojectm and projectm-sdl-cpp)

fgaz opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fgaz committed:
libprojectm: init at 4.1.4

fgaz committed:
projectm-sdl-cpp: init at 0-unstable-2024-08-07

fgaz committed:
projectm: move to projectm_3

ProjectM 4 was split into libprojectm and projectm-sdl

Irgendeinwer commented:
1. [libprojectM v4.1.4](https://github.com/projectM-visualizer/projectm/releases/tag/v4.1.4) got released

2. Wouldn't the [sdl-frontend using Rust and the projectM crate](https://github.com/projectM-visualizer/frontend-sdl-rust) be the better solution for the projectm-sdl package?

fgaz commented:
> [libprojectM v4.1.4](https://github.com/projectM-visualizer/projectm/releases/tag/v4.1.4) got released

Updated

> Wouldn't the [sdl-frontend using Rust and the projectM crate](https://github.com/projectM-visualizer/frontend-sdl-rust) be the better solution for the projectm-sdl package?

Maybe? I just renamed this one to projectm-sdl-cpp to avoid confusion

tobim commented on pkgs/top-level/aliases.nix:
```diff
@@ -1117,6 +1117,7 @@ mapAliases {
   poac = cabinpkg; # Added 2025-01-22
   poretools = throw "poretools has been removed from nixpkgs, as it was broken and unmaintained"; # Added 2024-06-03
   powerdns = pdns; # Added 2022-03-28
+  projectm = throw "Since version 4, 'projectm' has been split into 'libprojectm' (the library) and 'projectm-sdl' (the SDL2 frontend). ProjectM 3 has been moved to 'projectm_3'"; # Added 2024-11-10
```
```suggestion
  projectm = throw "Since version 4, 'projectm' has been split into 'libprojectm' (the library) and 'projectm-sdl-cpp' (the SDL2 frontend). ProjectM 3 has been moved to 'projectm_3'"; # Added 2024-11-10
```

keenanweaver commented:
``` 
$ nix build --file /nix/store/qmwhvwkihb10wm18j0z3lx4pn4q3lccz-nixpkgs-review-3.1.0/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/keenan/.cache/nixpkgs-review/pr-355064/nixpkgs nixpkgs-overlays=/tmp/tmpaxa__38t' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr local-system x86_64-linux --argstr nixpkgs-path /home/keenan/.cache/nixpkgs-review/pr-355064/nixpkgs --argstr nixpkgs-config-path /tmp/tmpecavw10h.nix --argstr attrs-path /home/keenan/.cache/nixpkgs-review/pr-355064/attrs.nix
error: builder for '/nix/store/yq7lsq0xc80a8nfwhizk4ji4cza97b32-projectm-sdl-cpp-0-unstable-2024-08-07.drv' failed with exit code 1;
       last 37 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/vs84wchrjhrx5wvjskdfk02si6jr057d-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > fixing cmake files...
       > cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/share/doc/projectMSDL -DCMAKE_INSTALL_INFODIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/include -DCMAKE_INSTALL_SBINDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/ii75mhh7sxl11167m1b86p0qrjsjyjmd-gcc-wrapper-14-20241116/bin/strip -DCMAKE_RANLIB=/nix/store/ii75mhh7sxl11167m1b86p0qrjsjyjmd-gcc-wrapper-14-20241116/bin/ranlib -DCMAKE_AR=/nix/store/ii75mhh7sxl11167m1b86p0qrjsjyjmd-gcc-wrapper-14-20241116/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/wq3n69p1liw8gmpcy8rn728p5mgqxmgi-projectm-sdl-cpp-0-unstable-2024-08-07 -DENABLE_DESKTOP_ICON:BOOL=FALSE
       > -- The C compiler identification is GNU 14.2.1
       > -- The CXX compiler identification is GNU 14.2.1
       > -- Detecting C compiler ABI info
       > -- Detecting C compiler ABI info - done
       > -- Check for working C compiler: /nix/store/ii75mhh7sxl11167m1b86p0qrjsjyjmd-gcc-wrapper-14-20241116/bin/gcc - skipped
       > -- Detecting C compile features
       > -- Detecting C compile features - done
       > -- Detecting CXX compiler ABI info
       > -- Detecting CXX compiler ABI info - done
       > -- Check for working CXX compiler: /nix/store/ii75mhh7sxl11167m1b86p0qrjsjyjmd-gcc-wrapper-14-20241116/bin/g++ - skipped
       > -- Detecting CXX compile features
       > -- Detecting CXX compile features - done
       > -- Found OpenGL: /nix/store/4yhzwb09jja7wfifhh0qv9d671gf0g8c-libglvnd-1.7.0/lib/libOpenGL.so
       > -- Found ZLIB: /nix/store/bb9h0qjkl5fh6yr48lw3930gdwvackh5-zlib-1.3.1/lib/libz.so (found version "1.3.1")
       > -- Found PCRE2: /nix/store/1ibgqrs6r9s5fx060k8siq0flgysp28a-pcre2-10.44/lib/libpcre2-8.so
       > CMake Error at /nix/store/01dapiw4pjrc9s0166dcyg9xkw0rv01z-cmake-3.31.5/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
       >   Could NOT find Utf8Proc (missing: UTF8PROC_LIBRARY UTF8PROC_INCLUDE_DIR)
       > Call Stack (most recent call first):
       >   /nix/store/01dapiw4pjrc9s0166dcyg9xkw0rv01z-cmake-3.31.5/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
       >   /nix/store/3a0wmmlas8cfjv5cr9rnzk07v984w243-poco-1.14.1-dev/lib/cmake/Poco/FindUtf8Proc.cmake:49 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
       >   /nix/store/01dapiw4pjrc9s0166dcyg9xkw0rv01z-cmake-3.31.5/share/cmake-3.31/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
       >   /nix/store/3a0wmmlas8cfjv5cr9rnzk07v984w243-poco-1.14.1-dev/lib/cmake/Poco/PocoFoundationConfig.cmake:6 (find_dependency)
       >   /nix/store/01dapiw4pjrc9s0166dcyg9xkw0rv01z-cmake-3.31.5/share/cmake-3.31/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
       >   /nix/store/3a0wmmlas8cfjv5cr9rnzk07v984w243-poco-1.14.1-dev/lib/cmake/Poco/PocoJSONConfig.cmake:2 (find_dependency)
       >   /nix/store/3a0wmmlas8cfjv5cr9rnzk07v984w243-poco-1.14.1-dev/lib/cmake/Poco/PocoConfig.cmake:29 (find_package)
       >   CMakeLists.txt:76 (find_package)
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/yq7lsq0xc80a8nfwhizk4ji4cza97b32-projectm-sdl-cpp-0-unstable-2024-08-07.drv'.
error: 1 dependencies of derivation '/nix/store/mpvywfh2hvf183dz13na4kwqa4qi5iw5-review-shell.drv' failed to build
```

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355064`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>projectm-sdl-cpp</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libprojectm</li>
    <li>projectm_3</li>
  </ul>
</details>

fgaz commented:
Fixed

keenanweaver reviewed (approved):
Thanks for updating this!

![image](https://github.com/user-attachments/assets/4574797b-6878-4f14-9bde-624c856c5e35)
![image](https://github.com/user-attachments/assets/38fa149a-675a-4a16-a220-eaedbac3edd8)

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355064`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>libprojectm</li>
    <li>projectm-sdl-cpp</li>
    <li>projectm_3</li>
  </ul>
</details>
